protovalidate/internal/extra_func.py

```diff
@@ -380,10 +378,7 @@ def __prefix_length(self) -> bool:
 
         start = self._index
 
-        while True:
-            if self._index >= len(self._string) or not self.__digit():
```

Comment:
Noticed this pattern in a few places - I think it is better to lift the condition into the while statement if possible. We were doing this already in a few places in the URI parsing but not here.